src/CMakeLists.txt

```
@@ -430,12 +431,39 @@ add_custom_command (OUTPUT gmt_dimensions.h
 # autogenerated files target
 add_custom_target (gen_gmt_headers DEPENDS ${GMT_GEN_HDRS})
 
+# generate gmt_moduleinfo.h
+set (_moduleinfo)
+foreach (_prog_src ${GMT_PROGS_SRCS} begin.c clear.c end.c figure.c inset.c subplot.c)
+	file (READ ${_prog_src} _src_content)
+	string (REGEX MATCH "#define THIS_MODULE_CLASSIC_NAME[ \t]*\"([^\n]*)\"\n" _this_module_classic_name ${_src_content})
+	set (_this_module_classic_name ${CMAKE_MATCH_1})
+	string (REGEX MATCH "#define THIS_MODULE_MODERN_NAME[ \t]*\"([^\n]*)\"\n" _this_module_modern_name ${_src_content})
+	set (_this_module_modern_name ${CMAKE_MATCH_1})
+	string (REGEX MATCH "#define THIS_MODULE_LIB[ \t]*\"([^\n]*)\"\n" _this_module_lib ${_src_content})
+	set (_this_module_lib ${CMAKE_MATCH_1})
+	string (REGEX MATCH "#define THIS_MODULE_PURPOSE[ \t]*\"([^\n]*)\"\n" _this_module_purpose ${_src_content})
+	set (_this_module_purpose ${CMAKE_MATCH_1})
+	string (REGEX MATCH "#define THIS_MODULE_KEYS[ \t]*\"([^\n]*)\"\n" _this_module_keys ${_src_content})
+	set (_this_module_keys ${CMAKE_MATCH_1})
+	list (APPEND _moduleinfo "\t{\"${_this_module_modern_name}\", \"${_this_module_classic_name}\", \"${_this_module_lib}\", \"${_this_module_purpose}\", \"${_this_module_keys}\"},")
+endforeach (_prog_src ${GMT_PROGS_SRCS})
+# Two more entries for "gmtread" and "gmtwrite"
+list (APPEND _moduleinfo "\t{\"gmtread\", \"gmtread\", \"core\", \"Read GMT objects into external API\", \"-T-,<?{,>?}\"},")
+list (APPEND _moduleinfo "\t{\"gmtwrite\", \"gmtwrite\", \"core\", \"Write GMT objects from external API\", \"-T-,<?{,>?}\"},")
+list (JOIN _moduleinfo "\n" _moduleinfo)
+file (WRITE "${CMAKE_CURRENT_BINARY_DIR}/gmt_moduleinfo.h" ${_moduleinfo})
+
+
 # Configure header files to pass some of the CMake settings to the source code
 configure_file (gmt_config.h.in gmt_config.h)
 configure_file (psl_config.h.in psl_config.h)
 configure_file (gmt_version.h.in gmt_version.h)
 configure_file (isogmt.in isogmt${GMT_INSTALL_NAME_SUFFIX} @ONLY)
 configure_file (gmt-config.in gmt-config${GMT_INSTALL_NAME_SUFFIX} @ONLY)
+configure_file (gmt_glue.h.in gmt_glue.h)
+configure_file (gmt_glue.c.in gmt_glue.c)
+configure_file (gmt_suppl_glue.h.in gmt_suppl_glue.h)
+configure_file (gmt_suppl_glue.c.in gmt_suppl_glue.c)
```

```suggestion
set (SUPPL_NAME suppl)
configure_file (gmt_suppl_glue.h.in gmt_suppl_glue.h)
configure_file (gmt_suppl_glue.c.in gmt_suppl_glue.c)
```
I think we have to reset SUPPL_NAME.